src/functions/projectConfig.ts

```diff
@@ -0,0 +1,50 @@
+import { FunctionsConfig, FunctionsSingle } from "../firebaseConfig";
+import { FirebaseError } from "../error";
+
+export type NormalizedConfig = [FunctionsSingle, ...FunctionsSingle[]];
```

Comment:
So, it's `FunctionsSingle[]`? Why is a special config type needed?

Reply:
This type definition enforces that there be at least one item in the array. Too hard to read?

Reply:
It's maybe a little weird, but I guess I can live with it hah.